part2/stages/Functions/install-main

```diff
@@ -591,7 +594,20 @@ create_efi_boot_entries()
 
 copy_to_esp()
 {
-    local ESP=$(sfdisk -l -q | grep "EFI System" | awk '{ print $1 }')
+    # TARGET_DISK not defined in upgrades; locate our
+    # installation disk manually.
+    if [ -z "${TARGET_DISK}" ]; then
+        local part="$(vgs xenclient --noheadings -o pv_name | sed -e 's/\s*//g' -e 's|^/dev/||g')"
+
+        # get disk from partition (e.g. sda2 --> sda). we could
+        # simply strip off the trailing digits, but that won't
+        # work for other disk formats (e.g. nvme). this method
+        # will get the /sys/device entry and use the parent dir
+        # as the disk name.
+        local TARGET_DISK="$(readlink /sys/class/block/$part | awk -F/ '{ print $(NF-1) }')"
```

Comment:
The installer already has [helpers](https://github.com/OpenXT/installer/blob/master/common/stages/Functions/library) to handle this kind of things:
```bash
local devnode="$(vgs xenclient --noheadings -o pv_name)"
local disk=$(get_devnode_disk "${devnode}")
local part=$(get_devnode_part "${devnode}")
```